packages/daemon/src/veramo/did-manager.ts

```diff
@@ -1,16 +1,45 @@
-import { agent, SERVICE_DID_ALIAS, DID_PROVIDER, KMS } from './index';
+import { createVeramoAgent, SERVICE_DID_ALIAS, DID_PROVIDER, KMS } from './index';
+import { IDataStore, IDIDManager, IKeyManager, IResolver, TAgent} from "@veramo/core";
+import { IDataStoreORM } from "@veramo/data-store";
+import { Connection } from "typeorm";
+import { VeramoAgentConfigOverrides } from "./veramoAgentConfig";
+import fs from "fs";
 
+// requires calling initVeramo to create this local singleton agent and dbConnection
+let agent: TAgent<IDIDManager & IKeyManager & IDataStore & IDataStoreORM & IResolver>;
+let dbConnection: Connection;
+let dbFile: string;
 /**
  * Truing up the local Veramo DID store.
  */
-export const initDids = async () =>
+export const initVeramo = async (overrides: VeramoAgentConfigOverrides = null) => {
+  dbFile = overrides?.dbFile;
+  ({ agent, dbConnection } = (await createVeramoAgent(overrides)));
   // Create our service's DID to sign stuff on our behalf (happens once per environment and is a no-op afterwards).
-  agent.didManagerGetOrCreate({
+  await agent.didManagerGetOrCreate({
     alias: SERVICE_DID_ALIAS,
     provider: DID_PROVIDER,
     kms: KMS,
   });
 
+  return agent;
+};
+
+/**
+ * A utility for integration tests. Don't use in production code!
+ */
+export const _clearVeramo = async () => {
+  try {
+    await (await dbConnection).dropDatabase();
```

Comment:
Why do you have to nest `await dbConnection` twice? Can't you await it once and store the result?